compiler/noirc_evaluator/src/ssa/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -1329,10 +1332,12 @@ impl AcirContext {
 
         // Optimistically try executing the brillig now, if we can complete execution they just return the results.
         // This is a temporary measure pending SSA optimizations being applied to Brillig which would remove constant-input opcodes (See #2066)
-        if let Some(brillig_outputs) =
-            self.execute_brillig(&generated_brillig.byte_code, &b_inputs, &outputs)
-        {
-            return Ok(brillig_outputs);
+        if attempt_execution {
```

Comment:
Can you add a comment here to say why this is needed?

Reply:
Done, and also where we're generating brillig for an unconstrained entrypoint.